tests/PHPUnit/PredisTestCase.php

```diff
@@ -204,31 +195,48 @@ protected function createClient(array $parameters = null, array $options = null,
     }
 
     /**
-     * Returns a base mocked connection from Predis\Connection\NodeConnectionInterface.
+     * Returns a basic mocked connection of the specified type.
      *
-     * @param mixed $parameters Optional parameters.
+     * @param string $interface  Connection type.
+     * @param mixed  $parameters Optional parameters.
      *
-     * @return mixed
+     * @return \Predis\Connection\NodeConnectionInterface
      */
-    protected function getMockConnection($parameters = null)
+    protected function getMockConnectionOfType($interface, $parameters = null)
     {
-        $connection = $this->getMock('Predis\Connection\NodeConnectionInterface');
+        $connection = $this->getMockBuilder($interface)->getMock();
+
+        // TODO: verify that $connection is an instance of \Predis\Connection\NodeConnectionInterface
```

Comment:
`$interface` can be checked with `is_a()`

Reply:
I was being lazy here and left the comment instead of implementing this check once and for all 🙄

PS: hmmm OK maybe saying "I was being lazy" after going through two merges with more than 300 conflicts and 483 files doesn't sound too convincing, but it's true 😂